### DISCUSSION TITLE
Auto-detected changed heading ID to replace all references in docusaurus project

### DISCUSSION THREAD

**Discussion by GretaSuper** [facebook/docusaurus#9519] [Ideas]
My project is to write the user guide for software tool. 
And we usually created different layes in main chapters like as following: 
**1 Command Name
   1.1 command1 syntax
   1.2 command2 syntax
         1.2.1 command 2 example
         Please refer to [Appendix A](Appendix_A).**

So there would be quite a lot of head reference link appearing in differenct chapters. 
Now we use heading ID i.e  **'### Hello World {#my-explicit-id}'** as unique heading ID for each page references.

   # 1 Command Name 
   ## 1.1 command1 syntax {#1.1} 
   ## 1.2 command2 syntax
   ### 1.2.1 command 2 example
   _Please refer to [1.1 command1 syntax](#1.1)._

We are asking if Docusaurus could provide more efficient way that we could only change the heading ID 
and you could also help replace the title name in brackets like
Previously it was **[1.1 command1 syntax] {#1.1}**, and we change to **1.1 software command1 syntax {#1.1}** 
So you could help replace all where it is refered to with  **[1.1 software command1 syntax] {#1.1}**

Thanks for your great work in Docusaurus!

**Comment by slorber** [comment]
I don't understand what you ask for sorry, please try to format your request better and make it clearer.


We have a syntax to let you write custom explicit heading ids: https://docusaurus.io/docs/markdown-features/toc#heading-ids
We don't provide a tool to let you refactor an anchor and edit multiple existing links at once. What could we provide exactly to solve this problem? 

We are also working on adding support for anchor links in our broken link checker: https://github.com/facebook/docusaurus/pull/9528
